src/Workspaces/Core/Portable/Rename/Renamer.RenameDocumentAction.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Immutable;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.Rename
+{
+    public static partial class Renamer
+    {
+        /// <summary>
+        /// Individual action from RenameDocument APIs in <see cref="Renamer"/>.
+        /// 
+        /// See <see cref="RenameDocumentActionSet" />
+        /// </summary>
+        public abstract class RenameDocumentAction
+        {
+            private readonly ImmutableArray<ErrorResource> _errorStringKeys;
+            protected OptionSet OptionSet { get; }
```

Comment:
What does it mean for each of these to be holding onto an OptionSet?

Reply:
Moving up, doesn't make a ton of sense to have them individually stored.